packages/dynamics-lib/src/client/entity-manager.js

```diff
@@ -89,55 +85,44 @@ export function retrieveMultipleAsMap (...entityClasses) {
     async () => retrieveMultipleFetchOperation(entityClasses),
     async data => {
       const optionSetData = await retrieveGlobalOptionSets().cached()
-      return data
-        .map((result, i) => result.value.map(v => entityClasses[i].fromResponse(v, optionSetData)))
-        .reduce((acc, val, idx) => {
-          acc[entityClasses[idx].definition.localCollection] = val
-          return acc
-        }, {})
+      return data.reduce((acc, item, idx) => {
+        acc[entityClasses[idx].definition.localCollection] = item.value.map(v => entityClasses[idx].fromResponse(v, optionSetData))
+        return acc
+      }, {})
```

Comment:
I appreciate you're just changing this, but it's not the easiest thing in the world to make sense of - may benefit from a little more expressiveness, as I don't really understand what you're getting here. Perhaps just break over several lines, assigning to some well named variables for us ordinary mortals?

Reply:
Yes, agreed.  Some additional comments might help to describe what the method is doing